hyperspy/misc/slicing.py

```diff
@@ -186,9 +186,14 @@ def _get_array_slices(self, slices, isNavigation=None):
                 array_slices.append(slice_)
         return array_slices
 
-    def _slicer(self, slices, isNavigation=None):
+    def _slicer(self, slices, isNavigation=None, out=None):
         array_slices = self._get_array_slices(slices, isNavigation)
-        _obj = self._deepcopy_with_new_data(self.data[array_slices])
+        if out is None:
+            _obj = self._deepcopy_with_new_data(self.data[array_slices])
+        else:
+            _obj = out
+            _obj.axes_manager = self.axes_manager.deepcopy()
+            _obj.data = self._data[array_slices]  # Changes data id!
```

Comment:
I actually worked on this in parallel (needed it for ROIs), and ended up with this block:
```python
if out is None:
    _obj = self._deepcopy_with_new_data(self.data[array_slices])
    for slice_, axis in zip(array_slices, _obj.axes_manager._axes):
        if (isinstance(slice_, slice) or
                len(self.axes_manager._axes) < 2):
            axis._slice_me(slice_)
        else:
            _obj._remove_axis(axis.index_in_axes_manager)
else:
    out.data = self.data[array_slices]
    _obj = out
    for slice_, axis_src, axis_dst in zip(
            array_slices, self.axes_manager._axes,
            out.axes_manager._axes):
        axis_src = axis_src.copy()
        if (isinstance(slice_, slice) or
                len(self.axes_manager._axes) < 2):
            axis_src._slice_me(slice_)
        axis_dst.update_from(axis_src)
```
Less reuse of the code, but it retains the `AxesManager` and all the axes. This will give an error if the number of axes changes, but from earlier discussions we agreed that it should be a requirement that the attributes of the axes might change, but that the axes themselves should remain the same.